tests/extension.spec.ts

```diff
@@ -104,4 +103,87 @@ describe("extension unit test", () => {
         extension.deactivate();
     });
 
+    context("activate with actionId as parameters in the URL", () => {
```

Comment:
Can you create needed tests in controller.spec.ts ? Unit tests should be in controller.spec.ts. Here it looks like integration test